### ISSUE TITLE
Support ES modules

### ISSUE THREAD

**Issue by lorensr** [apollographql/apollo-server#3786]
```js
import { ApolloServer } from 'apollo-server'
```

does not work in Node 13 with `"type": "module"` in `package.json`. More info here: https://github.com/apollographql/apollo-server/issues/1356#issuecomment-567101161

```
import { ApolloServer } from 'apollo-server'
         ^^^^^^^^^^^^
SyntaxError: The requested module 'apollo-server' does not provide an export named 'ApolloServer'
    at ModuleJob._instantiate (internal/modules/esm/module_job.js:92:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:107:20)
    at async Loader.import (internal/modules/esm/loader.js:164:24)
```

**Comment by derekjwilliams** [comment]
The workaround that I found (from https://github.com/apollographql/apollo-server/issues/1356) is

```
import apollo from 'apollo-server'
const { ApolloServer } = apollo
```

**Comment by platocrat** [comment]
> The workaround that I found (from #1356) is
> 
> ```
> import apollo from 'apollo-server'
> const { ApolloServer } = apollo
> ```

Using that suggested solution fails with the following implementation:

```javascript
const express = require('express')
import apollo from 'apollo-server'
const { ApolloServer ) = apollo

const typeDefs = require('./schema.js')
const resolvers = require('./resolvers')
const models = require('./models')

const server = new ApolloServer({
    typeDefs,
    resolvers,
    // Pass models as context to Apollo Server to give us access
    // to models from our resolvers.
    context: { models }
})

const app = express()
server.applyMiddleware({ app })

app.listen({ port: 4000 }, () =>
	   console.log(`🚀Server ready at http://localhost:4000${server.graphqlPath}`)
	  )
```

The following is the error:

```
SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:1116:16)
    at Module._compile (internal/modules/cjs/loader.js:1164:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47
```

**Comment by glasser** [comment]
I'm going to declare this out of scope for AS3 for now. Hopefully when this is needed we can do it in a way that's backwards-compatible and doesn't require a major version bump.

**Comment by trevorblades** [comment]
I've put together a small example showing that nodejs imports are working with AS3 in Node 14: https://codesandbox.io/s/suspicious-wave-9z3r8?file=/src/index.js

**Comment by glasser** [comment]
Hmm. While we didn't actually do any particular work to make ESM work, I agree with @trevorblades that it does seem to work, at least with Node 14!

The comments from @lorensr and @derekjwilliams predate Node 14.

@platocrat 's comment first is about what happens when you lack `"type": "module"`, and then is about a require in their own code, not in Apollo Server.

[This comment](https://github.com/apollographql/apollo-server/issues/1356#issuecomment-681313954) references an error message that [this comment](https://github.com/nodejs/node/issues/32137#issuecomment-742554710) suggests is resolved in Node v14.13 (and v12.20).  (I think [this PR](https://github.com/nodejs/node/pull/35249) is the relevant one.)

[This comment](https://github.com/apollographql/apollo-server/issues/1356#issuecomment-763782259) from @jaibatrik and [this comment](https://github.com/apollographql/apollo-server/issues/1356#issuecomment-890767484) from @mushketyk don't have enough information to reproduce.

I'm going to assume that this issue is now fixed, as long as you're on Node v14.13 or v12.20!  If anyone (esp @jaibatrik or @mushketyk) can demonstrate ESM imports not working with one of those versions of Node (perhaps start by forking @trevorblades 's sandbox above) then we can reopen it.

(Separately, it might be nice to publish `apollo-server` built as ESM itself, but I don't know if that actually helps with any specific use cases. For Apollo Client we're doing that so you can load directly from a site like esm.run, but that seems less relevant for a server project.)

**Comment by glasser** [comment]
Sure, we can make a new issue, but it would be great if that issue came with a clear example of something that doesn't work today.  (This issue was about `import` not working in ESM mode, which appears to be fixed!)
